projects/plugins/crm/includes/ZeroBSCRM.InternalAutomatorRecipes.php

```diff
@@ -55,6 +55,7 @@
 	#} WP Hook tie-ins (for Mike [and 3rd party developers!], mostly)
 	zeroBSCRM_AddInternalAutomatorRecipe('contact.new','zeroBSCRM_IA_NewCustomerWPHook',array());
 	zeroBSCRM_AddInternalAutomatorRecipe('contact.update','zeroBSCRM_IA_EditCustomerWPHook',array());
+	zeroBSCRM_AddInternalAutomatorRecipe( 'contact.status.update', 'zeroBSCRM_IA_EditCustomerWPHook', array() );
```

Comment:
> Oh, we are using the old trigger system. Can we maybe simplify this in a more generic WordPress way and add a do_action() in the contact DAL instead?

That was the debate we had in our last session, whether we end up going our own way or reusing the current, already centralized system until we revise it. But moving away from the old trigger system later and adding the WP hooks directly to the DAL is not a big deal. We don't have a clear picture here, so the decision was to be conservative and use this centralized location to emit the hooks if they already exist. In this case, we need to create a new one because we need the full object/data and the current hook only passes the contact ID. So if we already have the full contact data, we avoid another SQL query that passes the object/data directly.

> This way, other parts of the code can also respond to the action (not just our automation)?

Other parts of the code can react to the WP action, because from the DAL we call this old system to trigger the action, and this system calls `do_action` at the end.

**Summary:**

- I am in favor of creating our own system for automation and centralizing everything we need there. And that was more or less the conclusion, we will make in the end. 
- We started to reuse the old centralized system because it already exists and to move forward with creating/testing the first triggers. 
- We need a centralized system because we need to multicast the same event in different ways and avoid adding all the hooks in the DAL. For instance:
   - A contact was updated. The DAL emmit the action and notifies it to the trigger event, using `do_action` or calling to the new triggering system `$jpcrm_trigger_system->event( 'contact_updated', $contact_old, $contact_updated )`
   - Then, in the new triggering system, we cast the WP actions. One single event could imply multiple actions:
      - Always: `do_action( 'jpcrm_contact_updated', $contact_updated, $contact_old )`
      - If the email was changed: `do_action( 'jpcrm_contact_email_updated', $contact_updated, $old_email )`
      - If the status was changed: `do_action( 'jpcrm_contact_status_updated', $contact_updated, $prev_status )`
      - If the phone was changed: `do_action( 'jpcrm_contact_email_updated', $contact_updated )`
      - If a tag was added: `do_action( 'jpcrm_contact_tag_added', $contact_updated, $tag_added )`
      - If a tag was removed: `do_action( 'jpcrm_contact_tag_removed', $contact_updated, $tag_removed )`
      - If a task was added: `do_action( 'jpcrm_contact_task_added', $contact_updated, $task_added )`
      - If a task was removed: `do_action( 'jpcrm_contact_task_removed', $contact_updated, $task_removed )`
 - That way we don't overload DAL with more code lines and move the responsibility, and the triggering logic, to another actor, the triggering system.
 - Later, once we implement most of the triggers, we will have to migrate the old triggers to the new ones to avoid duplicated system and unused code. TBH I've not gone deeper into the old system.

Reply:
Karen and I discussed it, and it seems better to implement our centralized triggering system separately from the current one. We will build it from scratch, avoiding patches and hacks, and without the need to understand and adapt to the old system. Later, once we implement most of the core triggers, we could migrate the old one to the new one if it's necessary.

We need a centralized triggering system to be able to emit multiple WP actions from one original event as I exposed in the example above.